analyzer/src/test/kotlin/managers/utils/NodeSupportTest.kt

```diff
@@ -198,14 +204,14 @@ class NodeSupportTest : WordSpec() {
                 fun ProtocolProxyMap.mapSingleValuesToString() =
                     mapValues { (_, proxies) ->
                         val (proxy, authentication) = proxies.single()
-                        listOf(proxy.toString(), authentication?.userName, authentication?.password?.let { String(it) })
+                        listOf(proxy.toString(), authentication!!.userName, String(authentication.password))
```

Comment:
This change seems unrelated, and the simplification also is not correct, as `authentication` *can* be `null`. I was just missing a test case for that, but I've added it now: https://github.com/oss-review-toolkit/ort/pull/2750. That new est would result in an error with your change.

Reply:
The simplification was correct before your change, and was required because the type resolver had issues determining which matcher to use. But I have dropped all changes in this file from the last commit for that reason.

Reply:
Does this [commit](https://github.com/oss-review-toolkit/ort/pull/2750/commits/8bea7c2e855657a380d8bb0955fecad3d94527a5) I just added to my PR help in that regard?

Reply:
IIRC there were more issues than just the nullability, but I'm not sure. I'd suggest to give it a try as part of your commit.

Reply:
I've incorporated my changes to this PR and restored your changes to `NodeSupportTest.kt` in the last commit.